### PR TITLE
Tone curve editor enhancements

### PR DESCRIPTION
**Pull request by theres**
Hello, please take a look on this tiny enhancements for tonecurve iop. I found this quite helpful and not intrusive. Same changes can be done for basecurve. 
- draw node coordinates on tonecurve chart
- `Ctrl + mousemove` with tonecurve node will decrease change speed by multiply of `darkroom/ui/scale_precise_step_multiplier`  
- `Shift + mousemove` with tonecurve node will increase change speed by multiply of `darkroom/ui/scale_rough_step_multiplier`  
- mouse ~middle~ right button click on tonecurve node will delete single selected node (in case of border nodes, it will just reset it's position)

### CODE REVIEW

**Comment by LebedevRI** [comment]
That last `	clang-format` commit, don't do that.
Also, commit messages are rather useless, and that is very common for github :/

**Comment by theres** [comment]
Sure, how about simply sqash and rebase + force push with full description before merge? I like this way quite a bit.

**Comment by LebedevRI** [comment]
It formats the whole file, which is rather irrelevant for this pr
And it turns the diff into pure unreadable garbage.

**Comment by houz** [comment]
I'd prefer right click to delete a node so it works the same as masks and zone system.

**Comment by theres** [comment]
Sure!
- clang formats on irrelevant lines reverted
- right click instead of middle one  (why I don't do that in the first place...?)
- I'll improve commit messages & push with force if you will recognize this PR as merge-capable.

File: src/iop/tonecurve.c [hunk 1]
```diff
@@ -1293,19 +1306,35 @@ static gboolean dt_iop_tonecurve_motion_notify(GtkWidget *widget, GdkEventMotion
...
+  int old_m_y = c->mouse_y;
+  c->mouse_x = event->x - inset;
+  c->mouse_y = event->y - inset;
+
+  const float mx = CLAMP(c->mouse_x, 0, width) / (float)width;
+  const float my = 1.0f - CLAMP(c->mouse_y, 0, height) / (float)height;
+
+  float multiplier;
+  GdkModifierType modifiers = gtk_accelerator_get_default_mod_mask();
+  if((event->state & modifiers) == GDK_CONTROL_MASK)
```
Comment (LebedevRI) [inline src/iop/tonecurve.c]: Add the `<SHIFT>` modifier too.

File: src/iop/tonecurve.c [hunk 2]
```diff
@@ -1293,19 +1306,35 @@ static gboolean dt_iop_tonecurve_motion_notify(GtkWidget *widget, GdkEventMotion
...
+    multiplier = dt_conf_get_float("darkroom/ui/scale_precise_step_multiplier");
+  }
+  else
+  {
+    multiplier = 1;
+  }
 
-  const float mx = c->mouse_x / (float)width;
-  const float my = 1.0f - c->mouse_y / (float)height;
+  const float dx = (old_m_x - c->mouse_x) * multiplier / (float)width;
```
Comment (LebedevRI) [inline src/iop/tonecurve.c]: Something is rather broken with this math.
Without modifiers, when moving the point with mouse, it is clearly moving out-of-sync with the mouse.

File: src/iop/tonecurve.c [hunk 3]
```diff
@@ -1306,27 +1306,31 @@ static gboolean dt_iop_tonecurve_motion_notify(GtkWidget *widget, GdkEventMotion
...
   {
     multiplier = dt_conf_get_float("darkroom/ui/scale_precise_step_multiplier");
   }
+  else if((event->state & modifiers) == GDK_SHIFT_MASK)
+  {
+    multiplier = dt_conf_get_float("darkroom/ui/scale_rough_step_multiplier");
+  }
   else
   {
     multiplier = 1;
```
Comment (LebedevRI) [inline src/iop/tonecurve.c]: Oh, there is also `darkroom/ui/scale_step_multiplier`
Just copy-paste it from `_move_point_internal()` :)
Or, better idea, call that function from here.

**Comment by LebedevRI** [comment]
@theres
ok, desync seems to be gone.
But, did you change the way `draw node coordinates on tonecurve chart` are printed?
They look different now, in right-bottom corner (which always will have histogram, unlike top-left corner), with same dark color as the underlying histogram. Rather unreadable i'd say.

**Comment by theres** [comment]
Nop, I just add this that way. I've darken the color and change font weight to make this a bit more visible now. This is however matter of taste how it should look like so I'm open for suggestions. 

Btw. I've some PoC with moving this away to css (with gtk_render_* methods). However code is still far from perfect and I'm not sure if it's _the way_.

File: src/iop/tonecurve.c [hunk 4]
```diff
@@ -1212,6 +1203,28 @@ static gboolean dt_iop_tonecurve_draw(GtkWidget *widget, cairo_t *crf, gpointer
...
+    // draw information about current selected node
+    PangoLayout *layout;
+    PangoRectangle ink;
+    PangoFontDescription *desc = pango_font_description_copy_static(darktable.bauhaus->pango_font_desc);
+    pango_font_description_set_weight(desc, PANGO_WEIGHT_BOLD);
+    pango_font_description_set_absolute_size(desc, (DT_PIXEL_APPLY_DPI(0.06) * height) * PANGO_SCALE);
+    layout = pango_cairo_create_layout(cr);
+    pango_layout_set_font_description(layout, desc);
+
+    snprintf(text, sizeof(text), "%.2f / %.2f ( %+.2f)", tonecurve[c->selected].x * 100,
```
Comment (LebedevRI) [inline src/iop/tonecurve.c]: Ok, last thing.
For a/b channels, it is in `-128..128` range, not `0..100`

**Pull request closed without merging**